aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -1377,7 +1413,27 @@ AT_ERROR("svd: MAGMA library not found in "
 
   magma_int_t m = magma_int_cast(self.size(-2), "m");
   magma_int_t n = magma_int_cast(self.size(-1), "n");
-  auto k = std::min(m, n);
+  auto mn = std::min(m, n);
+
+  c10::Storage storage_rwork;
+  value_t* rwork = nullptr;
+
+  magma_int_t* iwork;
+  ALLOCATE_ARRAY(iwork, magma_int_t, 8 * mn);
+  // Copy-n-paste rwork size computation from BatchLinearAlgebra.cpp
```

Comment:
LAPACK <= 3.6 had a bug. CPU code has `lrwork = 7 * mn` to be compatible with older LAPACK versions. I believe this is not required for MAGMA and the more optimal value according to [the docs](https://icl.cs.utk.edu/projectsfiles/magma/doxygen/group__magma__gesdd.html) is `lrwork = 5 * k`.